### PR TITLE
Remove "Unknown" member from Affiliation Category, Add Testing and updates the DTO diagram accordingly

### PR DESCRIPTION
**Pull request by Steffengreiner**
**Description of changes**
This PR removes the `Unknown` member from the `Affliation Category` enum because determining an `Affiliation Category` is a crucial requirement in the offer generation process (pricing) and therefore has to be settled before the offer generation. 

**Misc** 
This will also fix the Issue in qoffer-2-portlet : https://github.com/qbicsoftware/qoffer-2-portlet/issues/70
Finally, This PR also updates the DTO Diagram accordingly and provides a Test for the Methods specified in the `Affiliation Category` enum. 
I'm unsure if that is really necessary, but it's better to be save than sorry 👍

### CODE REVIEW

File: src/test/groovy/life/qbic/datamodel/dtos/business/AffiliationCategorySpec.groovy [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        affiliationCategory.toString() == "internal"
+    }
+
+    def "Trying to create an Enum with a non-specified Category throws MissingPropertyException"() {
+        when:
+        AffiliationCategory affiliationCategory = AffiliationCategory.UNKNOWN
+
+        then:
+        thrown(MissingPropertyException)
+    }
```
Comment (jenniferboedker) [inline src/test/groovy/life/qbic/datamodel/dtos/business/AffiliationCategorySpec.groovy:28]: Does it make sense to test for something that is not there? I would remove this
```suggestion

```

**Review by jenniferboedker — DISMISSED** [review]
LGTM

File: src/test/groovy/life/qbic/datamodel/dtos/business/AffiliationCategorySpec.groovy (hunk 1 above)
Comment (Steffengreiner) [reply src/test/groovy/life/qbic/datamodel/dtos/business/AffiliationCategorySpec.groovy]: Good Point! It probably is unnecessary to test an Enum anyway 👍

**Review by jenniferboedker — APPROVED** [review]
LGMT

**Pull request merged**